app/assets/scripts/views/location/temporal-measurements.js

```diff
@@ -0,0 +1,213 @@
+        ? (sortedDayCount[day] = dayCount[day])
+        : (sortedDayCount[day] = 0)
+    );
+    return sortedDayCount;
+  };
+
+  const parseMonth = measurements => {
+    const monthOfYear = [
+      'JAN',
+      'FEB',
+      'MAR',
+      'APR',
+      'MAY',
+      'JUNE',
+      'JUL',
+      'AUG',
+      'SEP',
+      'OCT',
+      'NOV',
+      'DEC',
+    ];
+    const monthCount = measurements.reduce((prev, curr) => {
+      const month = monthOfYear[new Date(curr.date.local).getMonth()];
+      prev[month] = prev[month] ? prev[month] + 1 : 1;
+      return prev;
+    }, {});
+    let sortedMonthCount = {};
+    monthOfYear.forEach(month =>
+      monthCount[month]
+        ? (sortedMonthCount[month] = monthCount[month])
+        : (sortedMonthCount[month] = 0)
+    );
+    return sortedMonthCount;
+  };
+
+  if (fetching) {
+    return <LoadingMessage />;
+  } else if (error) {
+    return (
+      <ErrorMessage>
+        <p>We couldn&apos;t get any data.</p>
+        <InfoMessage>
+          <p>Please try again later.</p>
+          <p>
+            If you think there&apos;s a problem, please{' '}
+            <a href="mailto:info@openaq.org" title="Contact openaq">
+              contact us.
+            </a>
+          </p>
+        </InfoMessage>
+      </ErrorMessage>
+    );
+  }
+
+  return (
+    <Card
+      gridColumn={'1  / -1'}
+      renderHeader={() => (
+        <CardHeader className="card__header">
+          <TabbedSelector
+            tabs={parameters}
+            activeTab={activeTab}
+            onTabSelect={t => {
+              setActiveTab(t);
+            }}
+          />
+          <CardTitle>Temporal Coverage</CardTitle>
+        </CardHeader>
+      )}
+      renderBody={() => (
+        <div className="card__body">
+          {activeTabMeasurments.length ? (
+            <CardList gridTemplateColumns={'repeat(3, 1fr)'} className="inner">
+              <Card
+                renderBody={() => (
+                  <div className="card__body">
+                    <TemporalChart
+                      title="Hour of the Day"
+                      frequency={Object.values(hourCoverage)}
+                      xAxisLabels={Object.keys(hourCoverage)}
+                    />
+                  </div>
+                )}
+                renderFooter={() => null}
+              />
+              <Card
+                renderBody={() => (
+                  <div className="card__body">
+                    <TemporalChart
+                      title="Day of the Week"
+                      frequency={Object.values(dayCoverage)}
+                      xAxisLabels={Object.keys(dayCoverage)}
+                    />
+                  </div>
+                )}
+                renderFooter={() => null}
+              />
+              <Card
+                renderBody={() => (
+                  <div className="card__body">
+                    <TemporalChart
+                      title="Month of the Year"
```

Comment:
Are these sections of the data guaranteed to exist?

Reply:
Hm, I think they should - isn't it just a different temporal breakdown of the same data always?